operator/internal/webhook/admission/pcs/authorization/decoder.go

```diff
@@ -0,0 +1,127 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// */
+
+package authorization
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/NVIDIA/grove/operator/api/common/constants"
+	grovecorev1alpha1 "github.com/NVIDIA/grove/operator/api/core/v1alpha1"
+
+	groveschedulerv1alpha1 "github.com/NVIDIA/grove/scheduler/api/core/v1alpha1"
+	"github.com/go-logr/logr"
+	admissionv1 "k8s.io/api/admission/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/scale/scheme/autoscalingv1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var (
+	pcsGVK            = grovecorev1alpha1.SchemeGroupVersion.WithKind(constants.KindPodCliqueSet)
+	pcsgGVK           = grovecorev1alpha1.SchemeGroupVersion.WithKind(constants.KindPodCliqueScalingGroup)
+	pclqGVK           = grovecorev1alpha1.SchemeGroupVersion.WithKind(constants.KindPodClique)
+	podGVK            = corev1.SchemeGroupVersion.WithKind("Pod")
+	secretGVK         = corev1.SchemeGroupVersion.WithKind("Secret")
+	roleGVK           = rbacv1.SchemeGroupVersion.WithKind("Role")
+	roleBindingGVK    = rbacv1.SchemeGroupVersion.WithKind("RoleBinding")
+	serviceGVK        = corev1.SchemeGroupVersion.WithKind("Service")
+	serviceAccountGVK = corev1.SchemeGroupVersion.WithKind("ServiceAccount")
+	hpaV2GVK          = autoscalingv2.SchemeGroupVersion.WithKind("HorizontalPodAutoscaler")
+	hpaV1GVK          = autoscalingv1.SchemeGroupVersion.WithKind("HorizontalPodAutoscaler")
+	podgangGVK        = groveschedulerv1alpha1.SchemeGroupVersion.WithKind("PodGang")
+
+	errDecodeRequestObject  = errors.New("failed to decode request")
+	errUnsupportedOperation = errors.New("unsupported operation")
+)
+
+// requestDecoder decodes the admission requests.
+// It optimizes only gets PartialObjectMetadata for the resource
+// as there is no need to get the full resource.
+type requestDecoder struct {
+	decoder admission.Decoder
+}
+
+func newRequestDecoder(mgr manager.Manager) *requestDecoder {
+	return &requestDecoder{
+		decoder: admission.NewDecoder(mgr.GetScheme()),
+	}
+}
+
+func (d *requestDecoder) decode(logger logr.Logger, req admission.Request) (*metav1.PartialObjectMetadata, error) {
+	reqGVK := schema.GroupVersionKind{
+		Group:   req.Kind.Group,
+		Version: req.Kind.Version,
+		Kind:    req.Kind.Kind,
+	}
+	switch reqGVK {
+	case pcsgGVK, pclqGVK, podGVK, serviceAccountGVK, serviceGVK, secretGVK, roleGVK, roleBindingGVK,
+		hpaV2GVK, hpaV1GVK,
+		podgangGVK:
+		return d.decodeAsPartialObjectMetadata(req)
+	default:
```

Comment:
You will also get requests for `Scale` subresource. That will go to the default case where it will print `unknown GVK` but that is misleading because you know of the `Scale` subresource but we choose to ignore it, right?